jujugui/static/gui/src/app/user/user.js

```diff
@@ -30,6 +41,23 @@ const User = class User {
     this.sessionStorage = cfg.sessionStorage || sessionStorage;
     this.localStorage = cfg.localStorage || localStorage;
     this._external = cfg.externalAuth || null;
+    this.expiration = cfg.expiration || getExpirationDate();
+  }
+
+  get expiration() {
+    return new Date(this.sessionStorage.getItem('expiration'));
+  }
+
+  set expiration(val) {
+    this.sessionStorage.setItem('expiration', val);
+  }
+
+  _checkExpiration() {
+    const now = new Date();
+    if (now > this.expiration) {
+      this.localStorage.clear();
+      this.sessionStorage.clear();
```

Comment:
We shouldn't keep the controller/model credentials, as that's actually the determinant of if we're logged in--if we have those tokens, we'll auto login, which creates the whole session time mismatch thing we've presently got.